src/app/Balloon.App.ClamAv/src/lib/Cli.php

```diff
@@ -0,0 +1,161 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Balloon
+ *
+ * @author      Raffael Sahli <sahli@gyselroth.net>
+ * @copyright   Copryright (c) 2012-2017 gyselroth GmbH (https://gyselroth.com)
+ * @license     GPLv3 https://opensource.org/licenses/GPL-3.0
+ */
+
+namespace Balloon\App\ClamAv;
+
+use \Balloon\Filesystem\Node\File;
+use \Balloon\App\AbstractApp;
+use \Balloon\App\AppInterface;
+use \Xenolope\Quahog\Client as ClamAv;
+use \Xenolope\Quahog\Exception\ConnectionException as ClamAvConnectionException;
+use \Socket\Raw\Factory as SocketFactory;
+
+class Cli extends AbstractApp
+{
+    const FILE_INFECTED = 0;
+    const FILE_OK = 1;
+
+    /**
+    * Socket
+    *
+    * @var string
+    */
+    protected $socket = 'unix:///var/run/clamav/clamd.ctl';
+
+    /**
+    * Maximum Stream Size
+    *
+    * @var int
+    */
+    protected $max_stream_size = 26214400;
+
+    /**
+    * Aggressiveness
+    *
+    * @var int
+    */
+    protected $aggressiveness = 3;
+
+    /**
+     * Init
+     *
+     * @return bool
+     */
+    public function init(): bool
+    {
+        return true;
+    }
+
+    /**
+     * Set options
+     *
+     * @var Iterable $config
+     */
+    public function setOptions(?Iterable $config=null): AppInterface
+    {
+        if($config === null) {
+            return $this;
+        }
+
+        foreach($config as $option => $value) {
+            switch($option) {
+                case 'socket':
+                    $this->socket = (string)$value;
+                    break;
+                case 'max_stream_size':
+                    $this->max_stream_size = (int)$value;
+                    break;
+                case 'aggressiveness':
+                    if ((int)$value <= 3) {
```

Comment:
If you check an option here you should throw an Exception (Balloon\App\ClamAv\Exception) if the value is not correct